### DISCUSSION TITLE
When can we hope to have LLVM 18 and GCC 14?

### DISCUSSION THREAD

**Discussion by FunMiles** [Homebrew/discussions#5296] [Everyday usage] [answered]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

I am a C++ programmer and  impatiently wanting to use C++23's "deducing this" feature. This feature is not supported in the current versions of clang++ and g++ from homebrew. However, clang++ 18 and g++ 14 do support it.
Both have been out a few months now. Is there any hope to get those to replace llvm 17 and gcc 13 as defaults in homebrew?

**Comment by Bo98** [comment]
> Both have been out a few months now

GCC 14 is not out. LLVM 18 was released last month and is being worked on and will hopefully be finished soon.

**Reply by TejasSharma1729** [reply]
GCC 14 is out. It is visible in the beta version of Ubuntu 24.04, which I use via `multipass` --> `sudo do-release-upgrade -d` --> `sudo apt install gcc-14 g++-14`.

**Reply by SMillerDev** [reply]
> Note: GCC 14 has not been released yet, so this document is a work-in-progress.

https://gcc.gnu.org/gcc-14/changes.html

**Reply by TejasSharma1729** [reply]
OK so I am to assume that I use a "beta" or "experimental" version in the beta, which hopefully will become a "release" by the time the main 24.04 is out then.

**Reply by Bo98** [reply]
Ubuntu 24.04 has a WIP version of GCC 14 built from a snapshot (https://gcc.gnu.org/pub/gcc/snapshots/) because GCC 14 is planned to be out around the same time as Ubuntu 24.04 and due to various code freezes Ubuntu needed to start preparing early or they wouldn't be able to ship it at all. You would have seen the same with gcc-12 with the last LTS, 22.04.

You can see how GCC 14 is marked as "development" on the homepage: https://gcc.gnu.org. The first stable version, GCC 14.1, will probably be released late April/early May.

**Comment by FunMiles** [comment]
LLVM 18.1.4 is now in homebrew. Thanks to those who made it happen.

**Comment by FA555** [comment]
GCC 14.1 has just been released (<https://gcc.gnu.org/gcc-14/>, though documentation still remains 404 at this time).

Hoping to see it in Homebrew soon, and thanks in advance to everyone who will work on it.

**Comment by fxcoudert** [comment, marked as answer]
GCC 14.1.0 is now in Homebrew. As stated above, LLVM 18.1.4 has been available for three weeks now.
